app/controllers/addresses_controller.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class AddressesController < ApplicationController
+  def update
+    service = Service.find(params[:service_id])
+    address = Address.find(params[:id])
+
+    if !service.addresses.include?(address)
```

Comment:
Nit: Normally, if I have both the `if` and the `else` parts of an `if` statement, I prefer to remove any negations in the condition by swapping the `if` and `else` bodies. It's just a minor aesthetic thing in order to avoid double negatives.